src/pack.c

```diff
@@ -491,12 +488,12 @@ static int make_final_pack(struct packdata *pack)
 
 	/* tar the staging directory up */
 	LOG(NULL, "starting tar for pack", "%s: %i to %i", pack->module, pack->from, pack->to);
-	string_or_die(&tar, TAR_COMMAND " " TAR_PERM_ATTR_ARGS " --directory=%s/%s/%i_to_%i/ "
-					"--numeric-owner -Jcf %s/%i/pack-%s-from-%i.tar delta staged",
-		      packstage_dir, pack->module, pack->from, pack->to, staging_dir, pack->to,
-		      pack->module, pack->from);
-	ret = system(tar);
-	free(tar);
+	string_or_die(&param1, "--directory=%s/%s/%i_to_%i/", packstage_dir, pack->module, pack->from, pack->to);
+	string_or_die(&param2, "%s/%i/pack-%s-from-%i.tar", staging_dir, pack->to, pack->module, pack->from);
+	char *const tarcmd[] = { TAR_COMMAND, TAR_PERM_ATTR_ARGS_STRLIST, param1, "--numeric-owner", "-Jcf", param2, "delta", "staged", NULL };
+	ret = system_argv(tarcmd);
+	free(param1);
+	free(param2);
 	LOG(NULL, "finished tar for pack", "%s: %i to %i", pack->module, pack->from, pack->to);
 	/* FIXME: this is a hack workaround, needs diagnosed and removed */
 	if ((ret != 0) && (ret != 256)) {
```

Comment:
@jrguzman-intel I'm not sure about the historical meaning of the FIXME comment, so I would leave it for now.

That said, @bryteise is correct that the ret check needs to be updated. Since system_argv() returns the WEXITSTATUS value of the waitpid() exit code, the check should be (ret != 0) && (ret != 1).